src/jsCodegen/JavaScriptCompiledXPath.ts

```diff
@@ -2,19 +2,93 @@ export type ContextItemIdentifier = string;
 
 export type FunctionIdentifier = string;
 
+export enum CompiledResultType {
+	Value,
+	Function,
+	None,
+}
+
 export type PartiallyCompiledAstAccepted = {
 	code: string;
+	// What kind of code was generated.
+	generatedCodeType: GeneratedCodeType;
 	isAstAccepted: true;
 	// Contains variable (and function) declarations for the upper compiled
 	// scope.
 	variables?: string[];
 };
 
-export function acceptAst(code: string, variables?: string[]): PartiallyCompiledAstAccepted {
+export enum GeneratedCodeBaseType {
+	// A normal value: 5, "test", true, ...
+	Value,
+	// Defining a variable: const test = "test";
+	Variable,
+	// A statement: if (this) { do that; }
+	Statement,
+	// A function has been generated: function foo() { return true; }
+	Function,
+	// An iterator is returned.
+	Iterator,
+	// A sequence of values has been returned. This isn't used at the
+	// moment but will come in handy when we get to add sequence support.
+	Sequence,
```

Comment:
Not sure we will though: Sequences are just a complex wrapper around Iterators.

Reply:
Then it might be better to just remove that